drivers/lora/lora_basics_modem/lbm_common.c

```diff
@@ -289,33 +289,57 @@ int lbm_lora_recv(const struct device *dev, uint8_t *msg, uint8_t msg_len, k_tim
 	return ret;
 }
 
-int lbm_lora_recv_async(const struct device *dev, lora_recv_cb cb, void *user_data)
+int lbm_lora_recv_async(const struct device *dev, const struct lora_recv_async_callbacks *cb)
 {
 	const struct lbm_lora_config_common *config = dev->config;
 	struct lbm_lora_data_common *data = dev->data;
+	ral_irq_t irqs = RAL_IRQ_LORA;
 	ral_status_t status;
 
 	/* Cancel ongoing reception */
 	if (cb == NULL) {
+		if (data->rx_state.async.preamble_detected || data->rx_state.async.header_valid) {
+			/* Reset IRQ state */
+			(void)ral_set_dio_irq_params(&config->ralf.ral, irqs);
```

Comment:
IRQs should also be reset when starting synchronous RX so as to not end RX at healder_valid or preamble detected in that mode.